project_issue_task_sync/models/project_issue.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# © 2017 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, fields, models
+
+
+class ProjectIssue(models.Model):
+    _inherit = 'project.issue'
+
+
+    def get_changed_vals(self, task):
+        vals = {}
+        if len(task.issue_ids) ==1:
+            if task.stage_id != self.stage_id:
+                vals['stage_id'] = self.stage_id.id
+            if task.user_id != self.user_id:
+                vals['user_id'] = self.user_id.id
+        return vals
+
+
+    def set_issue_vals(self):
+        for this in self:
+            task = this.sudo().task_id
+            """
+            if the create/write is called by a task creation skip all 
+            syncing functions and do not try to create a task
+            if it is not called vy a task creation 
+            and the project is "sync enabled" create the associated 
+            task.
+            """
+            if (this.project_id.sync_tasks_issues and not 
+                   self.env.context.get('sync_operation', False)):
```

Comment:
but i think this is logically flawed anyways: If I set `sync_operation` to `True`, I'd expect the sync to happen. Now I need to set it to `False`. Probably you want to remove the `not` and default to `True`?